### ISSUE TITLE
[select] - Ensure any new UA defaults meet WCAG 2.2 target size (minimum)

### ISSUE THREAD

**Issue by scottaohara** [openui/open-ui#1026]
Opening this so we make sure to consider making the default styling for the select element adhere to the 24x24px target size minimum, per the new WCAG 2.2 requirement. While there is an exception for unstyled controls provided by the user agent to not meet this minimum... being new styling it'd be great to at least meet the _minimum_ bar.

The current select element (as rendered in Edge on Windows) is 26x19px (if a single character 'option' used as the default).  The select with no chosen option is a tiny 22x19px

<img  alt="default select vs select modified to be 24 by 24 pixels" src="https://github.com/openui/open-ui/assets/4152514/e0816ee5-ecef-45e5-af5f-c462625c551c">
The select on the left is the current select element's default rendering (again 22x19px), with the one on the right being modified to be a minimum 24x24px.

If the triggering element cannot be updated in sizing, so as to not be so different from the existing select element - then I'd at least hope we could make the options meet the minimum - as those are honestly more the concern being that they're 'touching' sibling targets that the rule was created for, so as to help users with motor disabilities have a higher success rate in choosing the correct target via pointer devices (mouse, touch, etc.).

While this issue is just about the current 'select' work, we should keep this in mind for any new interactive elements that might come out of open ui.

https://www.w3.org/WAI/WCAG22/Understanding/target-size-minimum.html

**Comment by lukewarlow** [comment]
Fully agree with this, other UI that comes to mind is the textarea resize handle which while I haven't checked I would all but guarantee doesn't mean the requirements.

**Comment by scottaohara** [comment]
oh yeh.... absolutely doesn't.  many default styles for form controls/their parts don't meet the minimum.  hence an exception had to be written...

**Comment by mfreed7** [comment]
Fully supportive of this proposal. The current approach (using `appearance:base-select` on the `<select>` to opt-in to new behavior) has some limitations around changing styling such as overall width/height. But I believe @josepharhar has figured out an implementation that would make it possible.

**Comment by josepharhar** [comment]
For the button this doesn't require any changes because the fallback button as I've implemented is already 24x24, but option elements are not big enough. I tried these styles:
```css
select option {
  min-inline-size: 24px;
  min-block-size: 24px;
}
```
However, now the text is at the top of the block. In order to make it centered, which I figure people would want, I'd have to make the option element a flex container like this:
```css
select option {
  min-inline-size: 24px;
  min-block-size: 24px;
  display: flex;
  align-items: center;
}
```
Or maybe it would be better to just increase the font size to be big enough to meet the requirement without setting other rules? I could use some guidance here.

I'm going to reopen this since it has "select" in the title, but I can also make another issue if people want.

**Comment by gregwhitworth** [comment]
@josepharhar has the doc been updated to reflect this req; if so can we close this?

**Comment by scottaohara** [comment]
@josepharhar anything that can contribute to the overall size would be 'fine' to use. whether that be a font increase, more default padding, line height.    (and by 'fine' i mean in regards to meeting the rule, what's fine to use for browser implementation might be another story)

**Comment by josepharhar** [comment]
We can center the text using padding-top and padding-bottom, but I don't know if there is a good way in CSS to say that they should be equal and contribute to making the option element have a minimum total size and keep the text in the center.
